src/transformers/models/xglm/modeling_xglm.py

```diff
@@ -709,8 +709,8 @@ def forward(
             # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
             encoder_attention_mask = _expand_mask(encoder_attention_mask, inputs_embeds.dtype, tgt_len=input_shape[-1])
 
-        # embed positions
-        positions = self.embed_positions(input_ids, inputs_embeds, past_key_values_length)
+        # embed positions, cast from float32 to `inputs_embeds.dtype`
+        positions = self.embed_positions(input_ids, inputs_embeds, past_key_values_length).to(inputs_embeds.dtype)
```

Comment:
position (and `embed_positions`'s weight) are in float32 even if we load the model in float16. Need this cast to make later LayerNorm layer work.

Reply:
I think the right fix would be to make sure the weights have the correct dtype, as the embedding layer is the biggest one, so the memory save is very important.